src/lib/Parser.hs

```diff
@@ -116,10 +116,10 @@ sourceBlock' :: Parser SourceBlock'
 sourceBlock' =
       proseBlock
   <|> topLevelCommand
-  <|> liftM declToModule (topDecl <* eolf)
-  <|> liftM declToModule (instanceDef  <* eolf)
-  <|> liftM declToModule (interfaceDef <* eolf)
-  <|> liftM (Command (EvalExpr Printed) . exprAsModule) (expr <* eol)
+  <|> fmap declToModule (topDecl <* eolf)
+  <|> fmap declToModule (instanceDef  <* eolf)
+  <|> fmap declToModule (interfaceDef <* eolf)
+  <|> fmap (Command (EvalExpr Printed) . exprAsModule) (expr <* eol)
```

Comment:
I'm pretty sure @dougalm has just changed those to `liftM` a few commits ago 😛

Reply:
Thanks for mentioning - I indeed found [@dougalm's changes to `liftM`](https://github.com/google-research/dex-lang/commit/599e2ee47c53b355954012c96e9515d3cbdc863c#diff-7352b0160734bf8ae9c7f0c87b1ff7641db108b16e448fa3393a1804ce6c1b3fR119-R122) in [`599e2e`](https://github.com/google-research/dex-lang/tree/599e2ee47c53b355954012c96e9515d3cbdc863c).

I reverted all changes renaming `liftM` to `fmap`. I also added a `.hlint.yaml` file with a `- ignore: {name: "Use fmap"}` directive to track this intentional coding style preference.

Reply:
Ha, yes, I probably did. I prefer `liftM` for cases like this because it's more specific than `fmap` and it signals "we're dealing with a monad here". To me `fmap` connotes "simple container" functors like lists. Even though monads *are* functors, we tend to use them in quite a different way from ordinary containers like lists, trees, etc.

Reply:
That makes sense!

Haskell neophyte question: I wonder if there are any meaningful differences (e.g. performance) between using `fmap` and `liftM`?

### Self-answer

[`liftM`'s definition](https://hackage.haskell.org/package/base-4.14.1.0/docs/src/GHC.Base.html#liftM): top-level function defined for `Monad m`.
```hs
-- | Promote a function to a monad.
liftM   :: (Monad m) => (a1 -> r) -> m a1 -> m r
liftM f m1              = do { x1 <- m1; return (f x1) }
```

[`fmap`'s definition](https://hackage.haskell.org/package/base-4.14.1.0/docs/src/GHC.Base.html#fmap): typeclass method.
```hs
class  Functor f  where
    fmap        :: (a -> b) -> f a -> f b
```

Monadic types seem to implement `Functor` via `fmap = liftM`:

```hs
newtype Identity a = Identity { runIdentity :: a }

instance Functor Identity where
    fmap = liftM
```

So calling `liftM` is actually more direct than calling `fmap` (which involves one extra indirection).